integration-libs/opf/base/components/opf-cta/opf-cta-scripts/opf-cta-scripts.service.ts

```diff
@@ -0,0 +1,245 @@
+      concatMap((ctaScriptsResponse: CtaScriptsResponse) => {
+        if (!ctaScriptsResponse?.value?.length) {
+          return throwError('Invalid CTA Scripts Response');
+        }
+        return of(
+          ctaScriptsResponse.value.map((ctaScript) => ctaScript.dynamicScript)
+        );
+      }),
+      take(1)
+    );
+  }
+
+  protected fillCtaScriptRequest() {
+    let paymentAccountIds: number[];
+
+    return this.getPaymentAccountIds().pipe(
+      concatMap((accIds) => {
+        paymentAccountIds = accIds;
+        return this.getScriptLocation();
+      }),
+      concatMap((scriptsLocation: CtaScriptsLocation | null) => {
+        return this.fillRequestForTargetPage(
+          scriptsLocation,
+          paymentAccountIds
+        );
+      })
+    );
+  }
+
+  protected fillRequestForTargetPage(
+    scriptsLocation: CtaScriptsLocation | null,
+    paymentAccountIds: number[]
+  ): Observable<CtaScriptsRequest> {
+    if (scriptsLocation === CtaScriptsLocation.PDP_QUICK_BUY) {
+      return this.fillCtaRequestforPDP(scriptsLocation, paymentAccountIds);
+    } else if (
+      scriptsLocation === CtaScriptsLocation.ORDER_HISTORY_PAYMENT_GUIDE ||
+      scriptsLocation === CtaScriptsLocation.ORDER_CONFIRMATION_PAYMENT_GUIDE
+    ) {
+      return this.fillCtaRequestforPagesWithOrder(
+        scriptsLocation,
+        paymentAccountIds
+      );
+    } else {
+      return throwError('Invalid Script Location');
+    }
+  }
+
+  protected fillCtaRequestforPagesWithOrder(
+    scriptLocation: CtaScriptsLocation,
+    paymentAccountIds: number[]
+  ): Observable<CtaScriptsRequest> {
+    return this.getOrderDetails(scriptLocation).pipe(
+      map((order) => {
+        const ctaScriptsRequest: CtaScriptsRequest = {
+          orderId: order?.code,
+          ctaProductItems: this.getProductItems(order as Order),
+          paymentAccountIds: paymentAccountIds,
+          scriptLocations: [scriptLocation],
+        };
+
+        return ctaScriptsRequest;
+      })
+    );
+  }
+
+  protected fillCtaRequestforPDP(
+    scriptLocation: CtaScriptsLocation,
+    paymentAccountIds: number[]
+  ) {
+    return this.currentProductService.getProduct().pipe(
+      filter(isNotNullable),
+      map((product: Product) => {
+        return {
+          orderId: undefined,
+          ctaProductItems: [{ productId: product?.code, quantity: 1 }],
+          paymentAccountIds: paymentAccountIds,
+          scriptLocations: [scriptLocation],
+        } as CtaScriptsRequest;
+      })
+    );
+  }
+
+  protected runCtaScripts(scripts: OpfDynamicScript[]) {
+    return from(scripts).pipe(
+      concatMap((script) => from(this.loadAndRunScript(script))),
+      reduce((loadedList: string[], script) => {
+        if (script?.html) {
+          loadedList.push(script.html);
+        }
+        return loadedList;
+      }, []),
+      map((list) => {
+        return this.removeScriptTags(list);
```

Comment:
remove all script tags so they are not displayed in DOM (we don't sanitize html snippet). the script tags are already executed in prior step 'loadAndRunScript' so not needed anymore.